IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo

```diff
@@ -2,6 +2,218 @@ within IDEAS.Examples.IBPSA;
+
+  final parameter Modelica.SIunits.MassFlowRate m_flow_turbulent(min=0)=deltaM * abs(m_flow_nominal)
+    "Turbulent flow if |m_flow| >= m_flow_turbulent";
+
+  IDEAS.Fluid.Actuators.Valves.TwoWayLinear val(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m_flow_nominal,
+    dpValve_nominal=dpValve_nominal,
+    allowFlowReversal=allowFlowReversal,
+    show_T=show_T,
+    from_dp=from_dp,
+    homotopyInitialization=homotopyInitialization,
+    linearized=linearized,
+    deltaM=deltaM,
+    rhoStd=rhoStd,
+    use_inputFilter=use_inputFilter,
+    riseTime=riseTime,
+    init=init,
+    y_start=y_start,
+    dpFixed_nominal=dpFixed_nominal,
+    l=l,
+    CvData=IDEAS.Fluid.Types.CvTypes.OpPoint) "Linear valve"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+
+  Modelica.Blocks.Interfaces.RealInput T(unit="K") "Temperature measurement" annotation (
+      Placement(transformation(
+        extent={{-20,-20},{20,20}},
+        rotation=270,
+        origin={0,106})));
+  Modelica.Blocks.Sources.RealExpression yExp(y=
+        IDEAS.Utilities.Math.Functions.smoothLimit(
+        (TSet - T)/P,
+        0,
+        1,
+        0.001))        "Control signal"
+    annotation (Placement(transformation(extent={{-94,30},{-54,50}})));
+  Modelica.Blocks.Interfaces.RealOutput y "Valve set point"
+    annotation (Placement(transformation(extent={{40,60},{60,80}}),
+        iconTransformation(extent={{40,60},{60,80}})));
+
+  IDEAS.Utilities.IO.SignalExchange.Overwrite oveVal(description="Opening valve signal. 0 means completely closed, 1 means completely open",
+      u(
+      min=0,
+      max=1,
+      unit="1")) "Overwrite valve actuator signal"
+    annotation (Placement(transformation(extent={{-38,30},{-18,50}})));
+  protected
+  parameter Medium.ThermodynamicState sta_default=
+     Medium.setState_pTX(T=Medium.T_default, p=Medium.p_default, X=Medium.X_default);
+  parameter Modelica.SIunits.DynamicViscosity eta_default=Medium.dynamicViscosity(sta_default)
+    "Dynamic viscosity, used to compute transition to turbulent flow regime";
+
+  final parameter Modelica.SIunits.MassFlowRate m_flow_nominal_pos = abs(m_flow_nominal)
+    "Absolute value of nominal flow rate";
+  final parameter Modelica.SIunits.PressureDifference dp_nominal_pos(displayUnit="Pa") = abs(dp_nominal)
+    "Absolute value of nominal pressure difference";
+
+equation
+  connect(val.y_actual, y)
+    annotation (Line(points={{5,7},{20,7},{20,70},{50,70}}, color={0,0,127}));
+  connect(port_a, val.port_a)
+    annotation (Line(points={{-100,0},{-56,0},{-10,0}}, color={0,127,255}));
+  connect(val.port_b, port_b)
+    annotation (Line(points={{10,0},{56,0},{100,0}}, color={0,127,255}));
+  connect(yExp.y, oveVal.u)
+    annotation (Line(points={{-52,40},{-40,40}}, color={0,0,127}));
+  connect(oveVal.y, val.y)
+    annotation (Line(points={{-17,40},{0,40},{0,12}}, color={0,0,127}));
+annotation (
```

Comment:
The annotation can be completely removed. Ideally the model is in the protected section (at the end of the model).

Reply:
I've moved the submodel to the protected section in https://github.com/open-ideas/IDEAS/pull/1111/commits/c483343398d29ecd7928df80854a5385fdc5c18e
but kept the annotation to allow graphical inspection of the submodel.